File: app_backend/app/schemas/factura.py
# app/schemas/factura.py
from pydantic import BaseModel, Field
from typing import Optional, List
from datetime import datetime
from .vendedor import Vendedor as VendedorSchema # Para anidar datos del vendedor
from .cliente import Cliente as ClienteSchema # Para anidar datos del cliente

class FacturaBase(BaseModel):
    numero_orden: Optional[str] = None
    numero_caso: Optional[str] = None
    honorarios_generados: float = Field(..., ge=0)
    gastos_generados: float = Field(..., ge=0)
    vendedor_id: int
    cliente_id: int

class FacturaCreate(FacturaBase):
    pass

class FacturaUpdate(BaseModel):
    numero_orden: Optional[str] = None
    numero_caso: Optional[str] = None
    honorarios_generados: Optional[float] = Field(None, ge=0)
    gastos_generados: Optional[float] = Field(None, ge=0)
    vendedor_id: Optional[int] = None
    cliente_id: Optional[int] = None

# --- CORRECCIÓN IMPORTANTE AQUÍ ---
# Este schema es el que se usa para las respuestas API y debe coincidir con el modelo
class Factura(FacturaBase): # Ya no hereda de InDBBase, se define explícitamente
    id: int
    # El campo se llama 'fecha_emision' en el modelo, no 'fecha_venta'
    fecha_emision: datetime 
    # El campo se llama 'created_at' en el modelo
    created_at: datetime

    vendedor: Optional[VendedorSchema] = None # Anidar objeto Vendedor
    cliente: Optional[ClienteSchema] = None # Anidar objeto Cliente

    class Config:
        from_attributes = True

class FacturasResponse(BaseModel):
    items: List[Factura]
    total_count: int

File: app_frontend/src/types/factura.ts
// src/types/factura.ts
import { Cliente } from './cliente';
import { Vendedor } from './vendedor';

export interface FacturaBase {
    numero_orden?: string | null;
    numero_caso?: string | null;
    honorarios_generados: number;
    gastos_generados: number;
    vendedor_id: number;
    cliente_id: number;
}

export interface FacturaCreate extends FacturaBase {}

export interface FacturaUpdate {
    numero_orden?: string;
    numero_caso?: string;
    honorarios_generados?: number;
    gastos_generados?: number;
    vendedor_id?: number;
    cliente_id?: number;
}

export interface Factura extends FacturaBase {
    id: number;
    fecha_emision: string; // El backend la convierte a string (ISO format)
    created_at: string;
    vendedor?: Vendedor; // Objeto anidado
    cliente?: Cliente;   // Objeto anidado
}

export interface FacturasResponse {
    items: Factura[];
    total_count: number;
}